### PR TITLE
Implement Lightgbm/Ray for inferencing with new RayScript helper class for AzureML integration

### PR DESCRIPTION
**Pull request by jfomhover**
In order to support Ray components, this PR introduces a new RayScript helper class that will let users run ray jobs in 3 setups: locally, remotely or in AzureML. This RayScript class is demonstrated by adding LightGBM/Ray for inferencing.

The approach taken here is NOT an AzureML native integration, but an experimental way to initialize a ray cluster from within an AzureML distributed job.

The RayScript class consists in:
- using DistributedComponent (mpi) to capture the number of nodes available
- running `ray start` cli on each node, connecting to the head node
- running the script `run()` method only on head node, while all cluster nodes just wait for the end of the job

### CODE REVIEW

File: src/scripts/inferencing/lightgbm_ray/score.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                delimiter=',',
+                newline='\n',
+                header='',
+                footer='',
+                comments='# ',
+                encoding=None
+            )
+
+
+def get_arg_parser(parser=None):
```
Comment (ruizhuanguw) [inline src/scripts/inferencing/lightgbm_ray/score.py:132]: When is this function needed?

File: conf/experiments/lightgbm-inferencing.yaml [hunk 2]
```diff
@@ -35,6 +35,7 @@ lightgbm_inferencing_config:
       build: docker/lightgbm-v3.2.1/linux_cpu_mpi_build.dockerfile
     - framework: lightgbm_c_api # v3.2.1 with C API prediction
       build: docker/lightgbm-custom/v321_patch_cpu_mpi_build.dockerfile
```
Comment (ruizhuanguw) [inline conf/experiments/lightgbm-inferencing.yaml:37]: Do we need a different build for lightgbm_ray?
Where is src/scripts/inferencing/lightgbm_ray/default.dockerfile be used?

File: src/scripts/inferencing/lightgbm_ray/score.py (hunk 1 above)
Comment (jfomhover) [reply src/scripts/inferencing/lightgbm_ray/score.py:132]: This function is not used here, but is used only during unit testing. I'm still relying on the shrike unit tests to ensure the arguments of this script match with the arguments provided in the yaml spec. This test provided by shrike required the module to have a main() function and a get_arg_parser() function.

https://github.com/microsoft/lightgbm-benchmark/blob/jfomhover/rayscripthelper/tests/aml/test_components.py

File: conf/experiments/lightgbm-inferencing.yaml (hunk 2 above)
Comment (jfomhover) [reply conf/experiments/lightgbm-inferencing.yaml:37]: the lightgbm_ray module contains its own default.dockerfile here https://github.com/microsoft/lightgbm-benchmark/blob/jfomhover/rayscripthelper/src/scripts/inferencing/lightgbm_ray/default.dockerfile

in this config file lightgbm-inferencing.yaml you can override this dockerfile with your own using this `build:` setting. but in the case of this lightgbm_ray component, we don't need to for now.

**Review by ruizhuanguw — APPROVED** [review]

**Pull request merged**